README.md

```diff
@@ -1,72 +1,38 @@
 # Code Base Investigator
-Code Base Investigator (CBI) is a tool designed to help developers reason about the use of _specialization_ (i.e. code written specifically to provide support for or improve performance on some set of platforms) in a code base.  Specialization is often necessary, but how a developer chooses to express it may impact code portability and future maintenance costs.
 
-The [definition of platform](https://doi.org/10.1016/j.future.2017.08.007) used by CBI is deliberately very flexible and completely user-defined; a platform can represent any execution environment for which code may be specialized.  A platform could be a compiler, an operating system, a micro-architecture or some combination of these options.
+[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.5018974.svg)](https://doi.org/10.5281/zenodo.5018974)
+[![OpenSSF Best Practices](https://www.bestpractices.dev/projects/8679/badge)](https://www.bestpractices.dev/projects/8679)
 
-## Code Divergence
-CBI measures the amount of specialization in a code base using [code divergence](http://doi.org/10.1109/P3HPC.2018.00006), which is defined as the arithmetic mean pair-wise distance between the code-paths used by each platform.
+Code Base Investigator (CBI) is an analysis tool that provides insight into the
+portability and maintainability of an application's source code.
 
-At the two extremes, a code divergence of 0 means that all of the platforms use exactly the same code, while a code divergence of 1 means that there is no code shared between any of the platforms.  The code divergence of real codes will fall somewhere in between.
+- Measure [code divergence](http://doi.org/10.1109/P3HPC.2018.00006) to
+  understand how much code is specialized for different compilers, operating
+  systems, hardware micro-architectures and more.
 
-## How it Works
-![Abstract Syntax Tree](./docs/example-ast.png)
+- Visualize the distance between the code paths used to support different
+  compilation targets.
 
-CBI tracks specialization in two forms: source files that are not compiled for all platforms; and regions of source files that are guarded by C preprocessor directives (e.g. `#ifdef`).  A typical run of CBI consists of a three step process:
-1) Extract source files and compilation commands from a configuration file or compilation database.
-2) Build an AST representing which source lines of code (LOC) are associated with each specialization.
-3) Record which specializations are used by each platform.
+- Identify stale, legacy, code paths that are unused by any compilation target.
```

Comment:
The latter, I think.  Code in a file that isn't touched by any platform in the configuration file gets associated with the empty platform set, and also called out as a percentage of SLOC:

```
---------------------------------------------
                      Platform Set LOC % LOC
---------------------------------------------
                                {}   2  4.88
                           {GPU 1}   1  2.44
                           {GPU 2}   1  2.44
                           {CPU 2}   1  2.44
                           {CPU 1}   1  2.44
                            {FPGA}  14 34.15
                    {GPU 2, GPU 1}   6 14.63
                    {CPU 1, CPU 2}   6 14.63
{FPGA, CPU 1, GPU 2, GPU 1, CPU 2}   9 21.95
---------------------------------------------
Code Divergence: 0.55
Unused Code (%): 4.88
Total SLOC: 41
```

We can't identify code we don't see at all.

Reply:
Maybe worth considering this as a feature in the future and have some routines that sniff the files to see if they could be code or not, and have a "lines of potentially stale code" value as well since unused code is still a maintenance burden.

Reply:
Great idea.  I've opened https://github.com/intel/code-base-investigator/issues/86 to track this.

The legacy interface (arguably) had better support for this, because the user was required to describe all the code in the codebase (via globs) in addition to providing the compile commands.  But it still wasn't perfect, because we couldn't identify source files outside of the globs.  Actually walking the source directory and alerting the user to anything we find would be much more useful.